loader/loader.go

```diff
@@ -58,6 +58,7 @@ type ObjectLoader interface {
 	GetAutoReleasePlansForApplication(c client.Client, ctx context.Context, application *applicationapiv1alpha1.Application) (*[]releasev1alpha1.ReleasePlan, error)
 	GetScenario(c client.Client, ctx context.Context, name, namespace string) (*v1beta1.IntegrationTestScenario, error)
 	GetAllEnvironmentsForScenario(c client.Client, ctx context.Context, integrationTestScenario *v1beta1.IntegrationTestScenario) (*[]applicationapiv1alpha1.Environment, error)
+	GetAllSnapshotsForBuildPipelineRun(c client.Client, ctx context.Context, pipelineRun *tektonv1.PipelineRun) (*[]applicationapiv1alpha1.Snapshot, error)
```

Comment:
You gonna hate me, but can we have ctx as a first param? According to https://pkg.go.dev/context ?

Reply:
I was thinking that we should handle that as part of [STONEINTG-774](https://issues.redhat.com/browse/STONEINTG-774) for consistency.